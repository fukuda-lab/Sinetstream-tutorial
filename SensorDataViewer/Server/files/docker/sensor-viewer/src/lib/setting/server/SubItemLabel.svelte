<script lang="ts">
  import { getContext } from "svelte";
  import { formKey, itemKey } from "./server";

  const { getParent } = getContext<Record<string, any>>(itemKey);
  const { targets } = getContext<Record<string, any>>(formKey);

  export let value: any = null;
</script>

<div class="row">
  <div />
  <div>
    {#if $targets.includes(getParent())}
      <i>check_circle</i>
    {:else}
      <i>unpublished</i>
    {/if}
  </div>
  <div>
    {#if value != null}
      <slot />: {JSON.stringify(value)}
    {:else}
      <slot />
    {/if}
  </div>
</div>
