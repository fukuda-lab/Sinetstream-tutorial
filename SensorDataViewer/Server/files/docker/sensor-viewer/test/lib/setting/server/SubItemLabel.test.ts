import { cleanup, render } from '@testing-library/svelte';
import { writable } from 'svelte/store';
import {
  afterEach, describe, expect, it,
} from 'vitest';

import SubItemLabel from '../../../../src/lib/setting/server/SubItemLabel.svelte';
import ItemContextTest, { DEFAULT_TEXT } from './ItemContextTest.svelte';

describe('SubItemLabel.svelt', () => {
  const PARENT = 'parent';
  const TEXT = 'child value';

  afterEach(() => cleanup());

  it('uncheck', () => {
    const targets = writable([]);
    const { container, getByText } = render(ItemContextTest, {
      props: {
        Component: SubItemLabel,
        value: PARENT,
        targets,
      },
    });
    expect(getByText('unpublished')).toBeTruthy();
    expect(() => getByText('check_circle')).toThrowError();
    expect(container.outerHTML).toMatchSnapshot();
  });

  it('check', () => {
    const targets = writable([PARENT]);
    const { container, getByText } = render(ItemContextTest, {
      props: {
        Component: SubItemLabel,
        value: PARENT,
        targets,
      },
    });
    expect(getByText('check_circle')).toBeTruthy();
    expect(() => getByText('unpublished')).toThrowError();
    expect(container.outerHTML).toMatchSnapshot();
  });

  it('value', () => {
    const targets = writable([PARENT]);
    const { container, getByText } = render(ItemContextTest, {
      props: {
        Component: SubItemLabel,
        value: PARENT,
        targets,
        props: {
          value: TEXT,
        },
      },
    });
    expect(getByText('check_circle')).toBeTruthy();
    expect(() => getByText('unpublished')).toThrowError();
    expect(getByText(`${DEFAULT_TEXT}: "${TEXT}"`)).toBeTruthy();
    expect(container.outerHTML).toMatchSnapshot();
  });

  it('slot', () => {
    const targets = writable([PARENT]);
    const { container, getByText } = render(ItemContextTest, {
      props: {
        Component: SubItemLabel,
        value: PARENT,
        targets,
        props: {
          text: TEXT,
        },
      },
    });
    expect(getByText('check_circle')).toBeTruthy();
    expect(() => getByText('unpublished')).toThrowError();
    expect(getByText(TEXT)).toBeTruthy();
    expect(() => getByText(DEFAULT_TEXT)).toThrowError();
    expect(container.outerHTML).toMatchSnapshot();
  });
});
