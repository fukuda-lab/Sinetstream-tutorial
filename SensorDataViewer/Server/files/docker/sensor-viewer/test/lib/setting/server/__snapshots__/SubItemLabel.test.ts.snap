// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubItemLabel.svelt > check 1`] = `"<body><div><div class=\\"row\\"><div></div> <div><i>check_circle</i></div> <div>default slot</div></div><!--<SubItemLabel>--><!--<ItemContextTest>--></div></body>"`;

exports[`SubItemLabel.svelt > slot 1`] = `"<body><div><div class=\\"row\\"><div></div> <div><i>check_circle</i></div> <div>child value</div></div><!--<SubItemLabel>--><!--<ItemContextTest>--></div></body>"`;

exports[`SubItemLabel.svelt > uncheck 1`] = `"<body><div><div class=\\"row\\"><div></div> <div><i>unpublished</i></div> <div>default slot</div></div><!--<SubItemLabel>--><!--<ItemContextTest>--></div></body>"`;

exports[`SubItemLabel.svelt > value 1`] = `"<body><div><div class=\\"row\\"><div></div> <div><i>check_circle</i></div> <div>default slot: \\"child value\\"</div></div><!--<SubItemLabel>--><!--<ItemContextTest>--></div></body>"`;
